### PR TITLE
Add documentation.

### PR DESCRIPTION
**Pull request by measure-fi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,100 +1,84 @@
...
-
-[Publishing](./Publishing.md) contains useful information on how to publish your contract
-to the world, once you are ready to deploy it on a running blockchain. And
-[Importing](./Importing.md) contains information about pulling in other contracts or crates
-that have been published.
-
-Please replace this README file with information about your specific project. You can keep
-the `Developing.md` and `Publishing.md` files as useful referenced, but please set some
-proper description in the README.
+# Cross-Chain AMM
```
Comment (measure-fi) [inline README.md:1]: This is largely borrowed from https://www.notion.so/volumefi/EPIC-08-12-Paloma-cross-chain-AMM-factory-c021f5b1d0964836bb12926672e00d6c

**Review by measure-fi — COMMENTED** [review]
```
#![warn(missing_docs)]
```

This needs to be added earlier in development for external facing code. A lot of what is included here are guesses as to the meaning, and even documentation stubs would go a long way towards making those guesses cogent.

There are a number of constructs like `(x0, y0, x1, y1)`. These are _usually_ better represented as `((x0, x1), (y0, y1))`. There are highly connected concepts (like token/amount) which are always separate in this code and it is a difficult interface and an easy spot for bugs. This may not be correctable at this point.

Types like `&[u8]` are very difficult to work with when they are being used for specific contexts. Struct definitions like `struct PoolId(&[u8])` would go a long way towards making this code understandable. Many of the field names (`chain0_id`, `chain1_id`) could probably be better named, and as part of a larger struct we could simply have `(source_chain, target_chain)` where each of those represented more than just an id.

Any line marked `XXX` needs a comment, I couldn't figure it out.

Any line marked `TODO` I will fix in a followup.

File: README.md [hunk 2]
```diff
@@ -1,100 +1,84 @@
...
+
+This is a template (factory) modeled after the Uniswap approach that allows anyone to create and deploy new
+cross-chain AMM pools on any chains which have a Compass contract deployed.
+
```
Comment (wc117) [inline README.md:5]:
```suggestion
This is a CW smart contract on Paloma to manage liquidity of Cross Chain AMM.
Liquidity providers need to deposit their tokens to the smart contracts on target chains to add liquidity on target chains.
Added liquidity put into queue first and convert it into liquidity later as follows.
Very first deposits should be done by initial depositors and the initial depositors are defined on creating pools.
When the first adding liquidity from both target chain, the ratio of the both tokens will be the first swapping ratio.
Deposits are pushed into the liquidity queue if the queue is empty or queued token is the same as the deposit token.
Current queued token is not the same as the deposit token, adds liquidity by token ratio in the pool and update the queue.
Liquidity is set to the address on paloma and the address can withdraw liquidity on both target chains anytime.
```

File: src/msg.rs [hunk 3]
```diff
@@ -1,69 +1,118 @@
...
+        /// Chain to transfer to.
         chain1_id: Uint256,
+        /// Token to transfor from.
         token0: String,
+        /// Token to transfer to.
         token1: String,
+        /// Receiver0 address.
         receiver0: String,
+        /// Receiver1 address.
         receiver1: String,
```
Comment (measure-fi) [inline src/msg.rs:85]: @wc117 I need clarity on what `receiver0`/`receiver1` means.

**Pull request merged**